### PR TITLE
docs: document hardware CI test jobs

### PR DESCRIPTION
**Pull request by mmilata**
Continuation of #1433.

### CODE REVIEW

File: docs/ci/jobs.md [hunk 1]
```diff
@@ -89,3 +89,27 @@ build? This is most likely it.**
...
+with whitelisted prefix. If you want hardware tests ran on your branch, make sure its
+name starts with `hardware/`.
+
+### hardware core btconly device test
+
+Also test on physical Trezor T but with Bitcoin-only firmware.
+
+### hardware legacy regular device test
+
+[Device tests](../tests/device-tests.md)  executed on physical Trezor 1.
```
Comment (tsusanka) [inline docs/ci/jobs.md:109]: Double space between link and "executed" :stuck_out_tongue:.

File: docs/ci/jobs.md [hunk 2]
```diff
@@ -89,3 +89,27 @@ build? This is most likely it.**
...
+
+### hardware core regular device test
+
+[Device tests](../tests/device-tests.md) that run against an actual physical Trezor T.
+The device needs to have special bootloader, found in `core/embed/bootloader_ci`, that
+makes it possible to flash firmware without confirmation on the touchscreen.
+
+All hardware tests are run nightly on the `master` branch, as well as on push to branches
+with whitelisted prefix. If you want hardware tests ran on your branch, make sure its
+name starts with `hardware/`.
```
Comment (tsusanka) [inline docs/ci/jobs.md:101]: It's `hw/` not `hardware/` isn't it? https://github.com/trezor/trezor-firmware/blob/master/ci/test-hw.yml#L12

**Review by tsusanka — COMMENTED** [review]
Great, two nits and it is good to go :)

File: docs/ci/jobs.md (hunk 1 above)
Comment (mmilata) [reply docs/ci/jobs.md]: Fixed in 4172d48.

File: docs/ci/jobs.md (hunk 2 above)
Comment (mmilata) [reply docs/ci/jobs.md]: D'oh, not like didn't write hw/mmilata/tt-tests a hundred times lately ... fixed in 4172d48.

**Review by tsusanka — APPROVED** [review]
:+1: squash and feel free to merge right away

**Pull request closed without merging**